### PR TITLE
Clean up the make targets

### PR DESCRIPTION
**Pull request by afbjorklund**
For #4928

### CODE REVIEW

**Comment by medyagh** [comment]
this is the autocomplete I see:

but what I am missing is the target to make out/docker-machine-driver-...


```
$ make drivers 
all                             drivers                         golint                          kvm-image                       minikube_iso                    test-iso
apt                             e2e-cross                       gvisor-addon-image              kvm_in_docker                   push-gvisor-addon-image         vet
buildroot-image                 e2e-darwin-amd64                install-hyperkit-driver         lint                            push-storage-provisioner-image  windows
check-release                   e2e-linux-amd64                 install-kvm-driver              lint-ci                         release-iso                     yum
checksum                        extract                         integration                     linux                           release-kvm-driver              
clean                           fmt                             integration-none-driver         linux-menuconfig                release-minikube                
cross                           gendocs                         integration-versioned           mdlint                          reportcard                      
cross-tars                      gocyclo                         iso-menuconfig                  minikube-darwin-amd64           storage-provisioner-image       
darwin                          gofmt                           iso_in_docker                   minikube-linux-amd64            test                            

```

and also the install-hyperkit doesn't work

```
$ make install-hyperkit-driver 
GOOS=darwin CGO_ENABLED=1 go build \
		-ldflags="-X k8s.io/minikube/pkg/drivers/hyperkit.version=v1.2.0 -X k8s.io/minikube/pkg/drivers/hyperkit.gitCommitID="83cdf00a1532292edfb56f0e644eed44b5f5bccf""   \
		-o out/docker-machine-driver-hyperkit k8s.io/minikube/cmd/drivers/hyperkit
sudo cp out/docker-machine-driver-hyperkit /Users/medya/bin/docker-machine-driver-hyperkit
Password: 
Sorry, try again.
Password: 
cp: /Users/medya/bin/docker-machine-driver-hyperkit: No such file or directory
make: *** [install-hyperkit-driver] Error 1
```

I do rather to have build docker-machine drivers without installing them.

**Comment by afbjorklund** [comment]
1) For me it is available as a `out/`, which expands if I start typing
  Maybe do the same kind of phony hack for the two drivers then ?

2) I'm not sure why it uses `$(HOME)/bin` and not `$(GOBIN)` ?
  Either way seems to be missing a `mkdir -p` on the directory.

**Comment by medyagh** [comment]
> 1. For me it is available as a `out/`, which expands if I start typing
>    Maybe do the same kind of phony hack for the two drivers then ?
> 2. I'm not sure why it uses `$(HOME)/bin` and not `$(GOBIN)` ?
>    Either way seems to be missing a `mkdir -p` on the directory.

for me it used to make the drivers in the out folder

File: Makefile [hunk 1]
```diff
@@ -230,30 +245,30 @@ ifeq ($(MINIKUBE_BUILD_IN_DOCKER),y)
...
 	@#golint: Json should be JSON (compat sed)
 	@sed -i -e 's/Json/JSON/' $@ && rm -f ./-e
 
 .PHONY: cross
-cross: out/minikube-linux-$(GOARCH) out/minikube-darwin-amd64 out/minikube-windows-amd64.exe
+cross: minikube-linux-amd64 minikube-darwin-amd64 minikube-windows-amd64.exe
 
 .PHONY: windows
-windows: out/minikube-windows-amd64.exe
+windows: minikube-windows-amd64.exe
```
Comment (tstromberg) [inline Makefile]: Why not in out/?

Comment (afbjorklund) [reply Makefile]: Oh, it is still created in out/. Just using the new (shorter) targets.

Comment (afbjorklund) [reply Makefile]: And the odd GOARCH's are still available, just not built by "cross".

**Comment by medyagh** [comment]
needs rebase

**Comment by afbjorklund** [comment]
No need to change the title, the label is enough. And I got the bot

**Comment by medyagh** [comment]
Sorry for the spam I was using the changing the title as a hack for myself to figure out the ones I looked at in one look ... But I realize that's caused annoyance.

**Pull request merged**